app/main.py

```diff
@@ -1,2 +1,56 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.capacity = 8
+        self.threshhold = int(self.capacity * 2 / 3)
+        self.length = 0
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.length == self.threshhold:
+            self.resize()
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, value, key_hash]
+                self.length += 1
+                break
+            if self.hash_table[index][0] == key and \
+                    self.hash_table[index][2] == key_hash:
+                self.hash_table[index][1] = value
+                break
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key):
+        key_hash = hash(key)
+        get_index = key_hash % self.capacity
+        cell = self.hash_table[get_index]
+        while True:
```

Comment:
I would probably try to extract the search logic shared between `__getitem__()` and `__setitem__()` into a new method.